cmd/query/app/querysvc/query_service.go

```diff
@@ -51,6 +54,9 @@ type QueryService struct {
 	options          QueryServiceOptions
 }
 
+// QSOption is the functional option for configuring QueryServiceOptions.
+type QSOption func(qOpts *QueryServiceOptions)
```

Comment:
Why do we need this as public?

Reply:
The original motivation was to use it here: https://github.com/jaegertracing/jaeger/blob/c5642b708be9e2577cc1c494889ae97946ccc78a/cmd/query/app/flags.go#L125

to avoid modifying the caller's signature if metrics querying is opted-out, but I think it's not a particularly large impact to simply add it as a parameter and pass in `nil` if opted-out, so I've removed this option for now.